syncoid

```diff
@@ -842,6 +843,20 @@ sub syncdataset {
 		pruneoldsyncsnaps($targethost,$targetfs,$newsyncsnap,$targetisroot,keys %{ $snaps{'target'}});
 	}
 
+	if (defined $args{'push-snap-removal'}) {
+		foreach my $snap ( sort { $snaps{'target'}{$a}{'creation'}<=>$snaps{'target'}{$b}{'creation'} } keys %{ $snaps{'target'} }) {
+			if (!exists $snaps{'source'}{$snap}) {
+				if ($targethost ne '') {
+					if ($debug) { print "$sshcmd $targethost $targetsudocmd $zfscmd destroy $targetfsescaped\@$snap\n"; }
+					system ("$sshcmd $targethost " . escapeshellparam("$targetsudocmd $zfscmd destroy $targetfsescaped\@$snap"));
+				} else {
+					if ($debug) { print "$targetsudocmd $zfscmd destroy $targetfsescaped\@$snap\n"; }
+					system ("$targetsudocmd $zfscmd snapshot $targetfsescaped\@$snap");
```

Comment:
This shouldn't snapshot the existing snapshot at the target but remove it :-) s/snapshot/destroy/

Reply:
Ah, yes, too much copy & paste.